### ISSUE TITLE
When `~/.config/helix/runtime` does *not* exist and `HELIX_RUNTIME` exists, `hx --grammar fetch` should use the second and not create the first

### ISSUE THREAD

**Issue by poliorcetics** [helix-editor/helix#9565]
I was very surprised when setting `HELIX_RUNTIME` did not allow me to move my grammars to another dir than `~/.config/helix/runtime` because the latter was recreated every time

## Details

Currently in the first runtime directory is the one where the grammars are always put when fetching them.

https://github.com/helix-editor/helix/blob/a1272bdb17a63361342a318982e46129d558743c/helix-loader/src/grammar.rs#L258-L263

But that directory is **always** `~/.config/helix/runtime` because that directory is pushed before `HELIX_RUNTIME` **even if it doesn't exist**

https://github.com/helix-editor/helix/blob/a1272bdb17a63361342a318982e46129d558743c/helix-loader/src/lib.rs#L52-L58

I think the more intuitive working would be `Fetching -> List possible runtime dirs -> Find first **existing** -> Fallback to first in list if not`

**Comment by the-mikedavis** [comment]
I wonder if we should just prioritize `HELIX_RUNTIME` over `~/.config/helix/runtime` in `helix_loader::prioritize_runtime_dirs`?

I think we don't currently prioritize it because packagers used it for immutable runtime files (like nixos) but now `HELIX_DEFAULT_RUNTIME` covers that use-case.

**Comment by poliorcetics** [comment]
Yes that would be my preferred choice too, but it's a breaking change and I'm unsure how much it would affect people

**Comment by poliorcetics** [comment]
Ah, this is a duplicate of https://github.com/helix-editor/helix/issues/6527, I missed it when first creating the issue. The situation has changed since the older was created though, I'm unsure if it's still fully relevant

**Comment by uciharis** [comment]
hi @poliorcetics , what OS do u use ? i am still strugle to setup the runtime bcos there is no runtime in the directory aka empty.

iam using opensuse and install helix via 1. zypper (opensuse package manager) 2. brew package manager, and both of them give the same result

**Comment by aaron-jack-manning** [comment]
Is this pending a decision about the design, or just requires implementing?

This is an issue I would be happy to contribute to if it's purely a question of resources and motivation.
